packages/compass-user-data/src/user-data.ts

```diff
@@ -103,14 +115,16 @@ export class UserData<T> {
     const absolutePath = await this.getFileAbsolutePath();
     const filePathList = await fs.readdir(absolutePath);
     const data = await Promise.allSettled(
-      filePathList.map((x) => this.readOne(x, options))
+      filePathList.map((x) =>
+        this.readAndParseFile(path.join(absolutePath, x), options)
+      )
     );
 
     const result = {
       data: [],
       errors: [],
     } as {
-      data: T[];
+      data: z.output<T>[];
       errors: Error[];
     };
```

Comment:
nit: This could just be a type, no need to actually `cast`, right?

i.e.

```js
const result: {
  data: z.output<T>[];
  errors: Error[];
} = {
  data: [],
  errors: [],
};
```

Reply:
made the suggested changes in [ce43d83](https://github.com/mongodb-js/compass/pull/4777/commits/ce43d83107db9d5fd43a26072beda316051a4c84)